src/main/java/com/redhat/rhjmc/containerjfr/platform/SelfDiscoveryPlatformClient.java

```diff
@@ -45,21 +45,28 @@
 import java.util.Collections;
 import java.util.List;
 
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+
 class SelfDiscoveryPlatformClient implements PlatformClient {
 
-    private static final ServiceRef VM_SELF_REF =
-            new ServiceRef("localhost", "This ContainerJFR", 0);
+    private static ServiceRef VM_SELF_REF;
     private final PlatformClient client;
 
     SelfDiscoveryPlatformClient(PlatformClient client) {
         this.client = client;
     }
 
     @Override
-    public List<ServiceRef> listDiscoverableServices() {
+    public List<ServiceRef> listDiscoverableServices(Logger logger) {
         List<ServiceRef> list = new ArrayList<>();
+        try {
+            VM_SELF_REF = new ServiceRef("localhost", "This ContainerJFR", 0);
+        } catch (Exception e) {
+            logger.info(e);
+            VM_SELF_REF = null;
```

Comment:
This doesn't seem right. Why would a `null` entry to added to the list in this case?

Reply:
should be fixed.